app/javascript/components/rooms/room/join/RoomJoin.jsx

```diff
@@ -55,8 +55,7 @@ export default function RoomJoin() {
 
   const methods = useForm(joinFormConfig);
 
-  const location = useLocation();
-  const path = encodeURIComponent(location.pathname);
+  const path = encodeURIComponent(document.location.pathname);
```

Comment:
`useLocation` seems to encourage determinism so it will always provide a relative path after a match (the basename will be filtered).
In such case and because of the API restrictions I guess there's no issues to use the `document.location` API.